packages/editor/src/CdcEditor.tsx

```diff
@@ -45,15 +45,15 @@ const CdcEditor: React.FC<WCMSProps> = ({ config: configObj, hostname, container
   const [state, dispatch] = useReducer(editorReducer, initialState)
 
   const setTempConfigAndUpdate = config => {
-    updateVizConfig(JSON.parse(JSON.stringify(config)))
+    updateVizConfig(_.cloneDeep(config))
     dispatch({ type: 'EDITOR_TEMP_SAVE', payload: config })
   }
 
   const resizeObserver = new ResizeObserver(entries => {
     const container = entries[0]
-    let { width } = container.contentRect
-    let newViewport = getViewport(width)
-    dispatch({ type: 'EDITOR_SET_VIEWPORT', payload: newViewport })
+    const { width } = container.contentRect
+    const newViewport = getViewport(width)
+    if (state.currentViewport !== newViewport) dispatch({ type: 'EDITOR_SET_VIEWPORT', payload: newViewport })
```

Comment:
@mpallansch @adamdoe adding this fix resolved the memory leak. If this won't cause any other issues I think I can clean up this PR and put it out of draft.

Reply:
I'm not sure how we didn't see that leak until now, this definitely seems necessary. I'm not seeing any issues with the addition.